File: Leetle/tests/reference_solutions/javascript/palindrome_number.js
/**
 * @param {number} x
 * @return {boolean}
 */
function isPalindrome(x) {
    if (x < 0) return false;
    const str = x.toString();
    return str === str.split('').reverse().join('');
}

// Main function to handle input from stdin
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim();
const x = parseInt(input);

const result = isPalindrome(x);
console.log(result.toString());

File: Leetle/tests/reference_solutions/java/PalindromeNumber.java
import java.util.*;

public class Main {
    public static boolean isPalindrome(int x) {
        if (x < 0) return false;
        String str = Integer.toString(x);
        String reversed = new StringBuilder(str).reverse().toString();
        return str.equals(reversed);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int x = Integer.parseInt(scanner.nextLine().trim());
        boolean result = isPalindrome(x);
        System.out.println(result);
    }
}
